File: cos20007  stuff/week10 stuff/10.1p task stuff/Clock/Clock.py
class Clock(object):
    __sec = Counter()
    __min = Counter()
    __hr = Counter()
    def __init__(self__sec,self__min,self__hr):
         self__sec = Counter()
         self__min = Counter()
         self__hr = Counter()

    def tick(self__sec,self__min,self__hr):
         self__sec = Counter().Increment
         if(self__sec.Counter().get_ticks >59):
            self__sec = Counter().Reset
            self__min = Counter().Increment
         
         if(self__min.Counter().get_ticks >59):
            self__min = Counter().Reset
            self__hr = Counter().Increment
         if(self__hr.Counter().get_ticks >23):
            self__hr = Counter().Reset
     
    def PrintTime(self__sec,self__min,self__hr):
          return String.Format("{0:D2}", self__hr) + ":" + String.Format("{0:D2}", self__min) + ":" + String.Format("{0:D2}", self__sec);
    
    def Reset(self__sec,self__min,self__hr):
          self__sec = Counter().Reset
          self__min = Counter().Reset
          self__hr = Counter().Reset

File: cos20007  stuff/week 3 stuff/3.2P stuff/CounterTest/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CounterTest
{
    public class Clock
    {
        private Counter _sec;
        private Counter _min;
        private Counter _hr;
        

        public Clock()
        {
            _sec = new Counter("sec");
            _min = new Counter("min");
            _hr = new Counter("hr");
           
        }

        public void tick()
        {
           
            _sec.Increment();
            if(_sec.Ticks >59)
            {
                _sec.Reset();
                _min.Increment();

            }

            if (_min.Ticks > 59)
            {
                _min.Reset();
                _hr.Increment();

            }

            if (_hr.Ticks > 23)
            {
                _hr.Reset();

            }
        }

        public string PrintTime()
        {
            

            return String.Format("{0:D2}", _hr.Ticks) + ":" + String.Format("{0:D2}", _min.Ticks) + ":" + String.Format("{0:D2}", _sec.Ticks);
        }

        public void reset()
        {
            _sec.Reset();
            _min.Reset();
            _hr.Reset();
        }


    }
}
